web/src/lib/components/Navbar.svelte

```diff
@@ -71,24 +104,44 @@
               return;
           }
 
-          if (u.nym === null || u.nym === "") {
-              showModal(TwitterUsername, true,
-                  (saved) => {
-                      if (saved) {
-                          if ($user && !$user.twitterUsernameVerified) {
-                              showModal(TwitterVerification, true);
-                          }
-                      } else {
-                          // trying to hide the modal if you didn't set your Twitter username logges you out
-                          logout(false);
-                      }
-                  });
-          }
+          if (u.nostrPublicKey === null) {
+            if (u.nym === null || u.nym === "") {
+                showModal(TwitterUsername, true,
+                    (saved) => {
+                        if (saved) {
+                            if ($user && !$user.twitterUsernameVerified) {
+                                showModal(TwitterVerification, true);
+                            }
+                        } else {
+                            // trying to hide the modal if you didn't set your Twitter username logges you out
+                            logout();
+                        }
+                    });
+            }
+        } else {
+            if (u.nym === null || u.nym === "") {
+                let gotProfile = false;
+                pool.connectAndGetProfile(decodeNpub(u.nostrPublicKey),
+                    nostrProfile => {
+                        if (gotProfile) {
+                            return;
+                        }
+                        gotProfile = true;
+
+                        let name = <string>nostrProfile.name;
+                        name = name.replace(/[^a-zA-Z0-9]/g, "").toLowerCase();
+                        while (name.length < 3) {
+                            name += "0"; // just pas with zeroes - not ideal, but they can always change it later
+                        }
+
+                        saveProfile(name, nostrProfile.picture);
+                    });
+            }
+        }
       });
   onDestroy(userUnsubscribe);
```

Comment:
When you unmount the component, you must unsubscribe from the subscriptions and close the relay connections. You can look at onDestroy in Chat.svelte, but it's basically:

```
        await pool.unsubscribeEverything();
        await pool.disconnect();
```

But you cannot use `onDestroy` here, because Navbar never destroys... The component gets reused for all the pages and Svelte optimizes it using the same one.